lib/vauth/digest.c

```diff
@@ -797,7 +797,8 @@ static CURLcode auth_create_digest_http_message(
   convert_to_ascii(hashbuf, ha2);
 
   if(digest->qop) {
-    hashthis = aprintf("%s:%s:%08x:%s:%s:%s", ha1, digest->nonce, digest->nc,
+    hashthis = aprintf("%s:%s:%08x:%s:%s:%s", ha1, digest->nonce,
+                       (unsigned int)digest->nc,
```

Comment:
This is an example of what I consider an "overeager" warning. I don't think %x is wrong with an integer variable.

Reply:
Felt the same for this one. The number of this case is low though. (Except in mbedtls.c where hex numbers and negated ones are used quite a bit in output.)